lib/spack/docs/packaging_guide.rst

```diff
@@ -2775,6 +2775,248 @@ Suppose a user invokes ``spack install`` like this:
 Spack will fail with a constraint violation, because the version of
 MPICH requested is too low for the ``mpi`` requirement in ``foo``.
 
+.. _package-attributes:
+
+------------------
+Package attributes
```

Comment:
Technically these are spec attributes (`self.spec.foo`), not package attributes (`self.spec.package.foo`)

Reply:
I suppose it's neither since their attributes on the `SpecBuildInterface`.  I could just change the section header to "Custom attributes", would that be sufficient?